# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/src/bindings.rs
# Commit: cc263e909c61
# Full Hash: cc263e909c61d2d9c7d7cc1ccb8d24c8d09bf5b7
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2020-03-19 03:53:34
# Regressor Bug: 1614655
# File Overlap Count: 2
# Description:
#   Bug 1614655 - Part 2: Allow 1:n mapping of Gecko - WR display items r=jrmuizel
#   
#   This patch changes the underlying storage for WR display items in DisplayItemCache
#   from Vec<Option<CachedDisplayItem> to Vec<Vec<CachedDisplayItem>>.
#   This allows storing multiple WebRender display items for one Gecko display item.
# ==============================================================================

diff -r 10897d6106a8 -r cc263e909c61 gfx/webrender_bindings/src/bindings.rs
--- a/gfx/webrender_bindings/src/bindings.rs	Wed Mar 18 17:08:49 2020 +0000
+++ b/gfx/webrender_bindings/src/bindings.rs	Wed Mar 18 17:09:51 2020 +0000
@@ -3519,8 +3519,8 @@
 }
 
 #[no_mangle]
-pub extern "C" fn wr_dp_start_item_group(state: &mut WrState, key: ItemKey) {
-    state.frame_builder.dl_builder.start_item_group(key);
+pub extern "C" fn wr_dp_start_item_group(state: &mut WrState) {
+    state.frame_builder.dl_builder.start_item_group();
 }
 
 #[no_mangle]
@@ -3529,7 +3529,10 @@
 }
 
 #[no_mangle]
-pub extern "C" fn wr_dp_finish_item_group(state: &mut WrState, key: ItemKey) -> bool {
+pub extern "C" fn wr_dp_finish_item_group(
+    state: &mut WrState,
+    key: ItemKey
+) -> bool {
     state.frame_builder.dl_builder.finish_item_group(key)
 }
 